Fix gtk support

Gaspard-- opened the pull request:
These fixes are based on swaywm/wlroots@755a1c9

I've separated the indentation fixes into a separate commit to aid reviewing.
I also refactored some local variables and if's into switch-cases.

Gaspard-- commented on source/protocol/WaylandServerProtocol.cpp:
```diff
@@ -0,0 +1,32 @@
+#include "protocol/WaylandServerProtocol.hpp"
+
+protocol::WaylandServerProtocol::WaylandServerProtocol()
+{
```
Seems like `wlProtocolLogger` isn't initialised. Also I think you should be using member initialisation, like:
```cpp
WaylandServerProtocol()
: wlDisplay(...),
[...]
```

Gaspard-- commented on source/main.cpp:
```diff
@@ -2,10 +2,37 @@
+static void debug_server_protocol(void *user_data, enum wl_protocol_logger_type direction, const struct wl_protocol_logger_message *message)
+{
+  (void)direction;
+  (void)message;
```
prefer not naming the parameters.

Gaspard-- commented on source/main.cpp:
```diff
@@ -2,10 +2,37 @@
+    {
+      if (!strncmp(argv[i], "--socket", 8))
+	{
+	  int socket_nb = atoi(argv[i] + 8);
```
You should be taking a socket name (not an actual socket) and pass that name to the wayland protocol so it creates the sockets.

plean replied:
that's exactly what I did
```
              printf("AddSocket method with name '%s' was %s\n", argv[i + 1], serverProtocol.AddSocket(argv[i + 1]) ? "unsuccessful" : "successful");
```
argv[i + 1] correspond to the socket's name

Gaspard-- replied:
ah, I misinterpreted the use of the `atoi`. Ok, works for me then.

Gaspard-- commented on include/protocol/WaylandServerProtocol.hpp:
```diff
@@ -0,0 +1,29 @@
+#include <vector>
+#include <string>
+#include <string.h>
+#include <errno.h>
```
Prefer moving includes not needed by the `.hpp` in the proper `.cpp`s instead (such as `errno.h` and `string.h`but not `string`). This avoids pulling stuff into other peoples code, and also speeds up compilation (even if slightly, it will add up).

Gaspard-- commented:
Note, this PR concerns #16

Gaspard-- commented:
TODO:
- [ ] repair main (so that you can do more than test the protocol)
- [ ] add socket name parameter for the client (`display::WaylandSurface`)
- [ ] start investigating the rest of the protocol

Gaspard-- commented:
Clean-up main:
- [ ] protocol active in all back-ends
- [ ] `WindowTree` accessible from protocol

Gaspard-- commented:
Ok, I'll merge and open appropriate issues then.

Gaspard-- committed:
Added fixes to gtk support.

These fixes are based on https://github.com/swaywm/wlroots/commit/755a1c9138d82b5184f36d5373dac7cf8fc642da

Gaspard-- committed:
Fix whitespace indentation
